src/ucp/wireup/address.c

```diff
@@ -863,13 +875,12 @@ ucs_status_t ucp_address_unpack(ucp_worker_t *worker, const void *buffer,
     }
 
     /* Unpack addresses */
-    address = address_list;
-    ptr     = aptr;
+    address   = address_list;
+    ptr       = aptr;
     dev_index = 0;
-    do {
-        if (*(uint8_t*)ptr == UCP_NULL_RESOURCE) {
-            break;
-        }
+
+    last_dev = (*(uint8_t*)ptr == UCP_NULL_RESOURCE);
```

Comment:
this check is not needed

Reply:
why? it has to be here to:
- not start the loop in case of there are no addresses (i.e. `UCP_NULL_RESOURCE` is set)
- stop the loop below when last address is processed (i.e. `UCP_ADDRESS_FLAG_LAST` is set)

Reply:
if no address you `goto out` on line 866

Reply:
> if no address you `goto out` on line 866

@brminich good point! thank you! fixed